app/backend/src/application/task/change-task-status.ts

```diff
@@ -0,0 +1,55 @@
+
+import StatusModel from 'database/models/status';
+import TaskModel from 'database/models/task';
+import { ChangeTaskStatusPayload } from 'types/transport/requests/task/change-task-status';
+import { ChangeTaskStatusResponsePayload } from 'types/transport/responses/task/change-task-status';
+import Mongoose from 'mongoose';
+
+/**
+ * Updates task status
+ *
+ * @param params - data required to change task status
+ */
+export async function changeTaskStatus(params: ChangeTaskStatusPayload): Promise<ChangeTaskStatusResponsePayload> {
+  const response: ChangeTaskStatusResponsePayload = { };
+  const task = await TaskModel.findById(params.taskId);
+
+  if (!task) {
+    return response;
+  }
+
+  // Remove taskId from previous status tasks array
+  if (task.statusId) {
+    const prevStatus = await StatusModel.findById(task.statusId);
+    const prevStatusTasks = (prevStatus?.tasks || []).filter(id => !(id as unknown as Mongoose.Types.ObjectId).equals(new Mongoose.Types.ObjectId(params.taskId)));
```

Comment:
hard to understand this line of code. It would be better to move it to the separate method with comments